contract_compassion/model/contracts.py

```diff
@@ -137,8 +143,10 @@ def _get_type(self):
         return [('O', _('General'))]
 
     @api.one
-    @api.depends('group_id.recurring_unit', 'group_id.recurring_value')
+    @api.depends('group_id.recurring_unit', 'group_id.recurring_value',
+                 'group_id.advance_billing_months')
     def _set_frequency(self):
+        recurring_value = 0
         frequencies = {
             '1 month': 'Monthly',

```

Comment:
You should wrap the texts Monthly, Bimonthly, ... inside the tranlsation function `_` so that it is translated to the language of the user. e.g. `_('Monthly')`